controllers/submariner/metrics_proxy_resources.go

```diff
@@ -0,0 +1,107 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package submariner
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/submariner-io/submariner-operator/api/v1alpha1"
+	"github.com/submariner-io/submariner-operator/controllers/constants"
+	"github.com/submariner-io/submariner-operator/controllers/helpers"
+	"github.com/submariner-io/submariner-operator/pkg/names"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// nolint:wrapcheck // No need to wrap errors here.
+func (r *Reconciler) reconcileMetricsProxyDaemonSet(instance *v1alpha1.Submariner, reqLogger logr.Logger) (*appsv1.DaemonSet,
+	error,
+) {
+	return helpers.ReconcileDaemonSet(instance, newMetricsProxyDaemonSet(instance, names.MetricsProxyComponent), reqLogger,
+		r.config.ScopedClient, r.config.Scheme)
+}
+
+func newMetricsProxyDaemonSet(cr *v1alpha1.Submariner, name string) *appsv1.DaemonSet {
+	labels := map[string]string{
+		"app":       name,
+		"component": "metrics",
+	}
+
+	daemonSet := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cr.Namespace,
+			Name:      name,
+			Labels:    labels,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{MatchLabels: map[string]string{
+				"app": name,
+			}},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "gateway-metrics-proxy",
+							Image:           getImagePath(cr, names.MetricsProxyImage, names.MetricsProxyComponent),
+							ImagePullPolicy: helpers.GetPullPolicy(cr.Spec.Version, cr.Spec.ImageOverrides[names.MetricsProxyComponent]),
+							Env: []corev1.EnvVar{
+								{Name: "NODE_IP", ValueFrom: &corev1.EnvVarSource{
+									FieldRef: &corev1.ObjectFieldSelector{
+										FieldPath: "status.hostIP",
+									},
+								}},
+							},
+							Command: []string{"/usr/bin/nc"},
+							Args:    []string{"-v", "-lk", "-p", fmt.Sprint(gatewayMetricsServerPort), "-e", "nc", "$(NODE_IP)", constants.MetricsGWPort},
+						},
```

Comment:
```suggestion
						*metricProxyContainer(cr, "gateway-metrics-proxy", gatewayMetricsServerPort, constants.MetricsGWPort),
```

(see suggestion further down)